core/templates/App.ts

```diff
@@ -255,46 +260,48 @@ angular.module('oppia').factory('$exceptionHandler', [
-          ].join('\n');
-          // To prevent an overdose of errors, throttle to at most 1 error every
-          // MIN_TIME_BETWEEN_ERRORS_MSEC.
-          if (
-            Date.now() - timeOfLastPostedError > MIN_TIME_BETWEEN_ERRORS_MSEC) {
-            // Catch all errors, to guard against infinite recursive loops.
-            try {
-              // We use jQuery here instead of Angular's $http, since the latter
-              // creates a circular dependency.
-              CsrfTokenService.getTokenAsync().then(function(token) {
-                $.ajax({
-                  type: 'POST',
-                  url: '/frontend_errors',
-                  data: $.param({
-                    csrf_token: token,
-                    payload: JSON.stringify({
-                      error: messageAndSourceAndStackTrace
-                    }),
-                    source: document.URL
-                  }, true),
-                  contentType: 'application/x-www-form-urlencoded',
-                  dataType: 'text',
-                  async: true
+      if (!DEV_MODE) {
+        sourceMappedStackTrace.mapStackTrace(
+          exception.stack, function(mappedStack) {
+            var messageAndSourceAndStackTrace = [
+              '',
+              'Cause: ' + cause,
+              exception.message,
+              mappedStack.join('\n'),
+              '    at URL: ' + window.location.href
+            ].join('\n');
+            var timeDifference = Date.now() - timeOfLastPostedError;
+            // To prevent an overdose of errors, throttle to at most 1 error
+            // every MIN_TIME_BETWEEN_ERRORS_MSEC.
+            if (timeDifference > MIN_TIME_BETWEEN_ERRORS_MSEC) {
+              // Catch all errors, to guard against infinite recursive loops.
+              try {
+                // We use jQuery here instead of Angular's $http, since the
+                // latter creates a circular dependency.
+                CsrfTokenService.getTokenAsync().then(function(token) {
+                  $.ajax({
+                    type: 'POST',
+                    url: '/frontend_errors',
+                    data: $.param({
+                      csrf_token: token,
+                      payload: JSON.stringify({
+                        error: messageAndSourceAndStackTrace
+                      }),
+                      source: document.URL
+                    }, true),
+                    contentType: 'application/x-www-form-urlencoded',
+                    dataType: 'text',
+                    async: true
+                  });
+
+                  timeOfLastPostedError = Date.now();
                 });
-
-                timeOfLastPostedError = Date.now();
-              });
-            } catch (loggingError) {
-              $log.warn('Error logging failed.');
+              } catch (loggingError) {
+                $log.warn('Error logging failed.');
+              }
             }
-          }
-        });
+          });
```

Comment:
Consider moving `);` to next line, to match opening indentation.

Reply:
Done.